src/core/Form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,482 @@
+
+  const toggleCalendar = (open: boolean, focus: boolean = false) => {
+    setCalendarVisible(open);
+
+    if (open) {
+      updateInputValueAsDate(inputRef.current?.value || null);
+    }
+
+    if (!open && focus) {
+      openButtonRef.current?.focus();
+    }
+  };
+
+  const conditionalSetInputValue = (newValue: string) => {
+    if (!('value' in props)) {
+      setInputValue(newValue);
+    }
+  };
+
+  const validateDatePickerValue = (newValue: string): Date | null => {
+    const date = dateAdapter.parse(newValue);
+    if (Number.isNaN(date.valueOf())) {
+      getLogger().warn(
+        `Date input value "${newValue}" could not be parsed to Date`,
+      );
+      return null;
+    }
+    if (!dayIsInRange(date, minDate, maxDate)) {
+      getLogger().warn(
+        `Date input value "${newValue}" is not within interval [minDate, maxDate]`,
+      );
+      return null;
+    }
+    return date;
+  };
+
+  const updateInputValueAsDate = (newValue: string | null) => {
+    if (newValue === null) {
+      setInputValueAsDate(newValue);
+    } else {
+      const date = validateDatePickerValue(newValue);
+      setInputValueAsDate(date);
+    }
+  };
+
+  const onDatePickerChange = (date: Date) => {
+    if (passProps.disabled) {
+      return;
+    }
+    const newValue = dateAdapter.format(date);
+    conditionalSetInputValue(newValue);
+    if (propOnChange) {
+      propOnChange({ value: newValue, date });
+    }
+  };
+
+  return (
+    <HtmlDiv
+      {...wrapperProps}
+      className={classnames(baseClassName, className, {
+        [dateInputClassNames.disabled]: !!passProps.disabled,
+        [dateInputClassNames.error]: status === 'error',
+        [dateInputClassNames.success]: status === 'success',
+        [dateInputClassNames.fullWidth]: fullWidth,
+        [dateInputClassNames.hasPicker]: datePickerEnabled,
+      })}
+    >
+      <HtmlSpan className={dateInputClassNames.styleWrapper}>
+        <Label
+          htmlFor={id}
+          labelMode={labelMode}
+          optionalText={optionalText}
+          className={classnames({
+            [dateInputClassNames.labelIsVisible]: labelMode !== 'hidden',
+          })}
+          tooltipComponent={tooltipComponent}
+        >
+          {labelText}
+        </Label>
+        <HintText id={hintTextId}>{hintText}</HintText>
+        <HtmlDiv className={dateInputClassNames.inputAndPickerWrapper}>
+          <HtmlDiv className={dateInputClassNames.inputElementContainer}>
+            <Debounce waitFor={debounce}>
+              {(debouncer: Function) => (
+                <HtmlInput
+                  autoComplete="off"
+                  {...passProps}
+                  value={'value' in props ? value : inputValue}
+                  id={id}
+                  className={dateInputClassNames.inputElement}
+                  forwardedRef={inputRef}
+                  placeholder={visualPlaceholder}
+                  {...{ 'aria-invalid': status === 'error' }}
```

Comment:
This logic could be altered so that there would not be a redundant `aria-invalid: false` in DOM by default.

Reply:
Removed aria-invalid when false ✔️